ui-testing-core/src/main/java/com/avito/android/test/page_object/TextElement.kt

```diff
@@ -0,0 +1,137 @@
+package com.avito.android.test.page_object
+
+import android.text.SpannableString
+import android.text.style.ClickableSpan
+import android.view.View
+import android.widget.TextView
+import androidx.test.espresso.PerformException
+import androidx.test.espresso.UiController
+import androidx.test.espresso.ViewAction
+import androidx.test.espresso.action.CoordinatesProvider
+import androidx.test.espresso.util.HumanReadables
+import com.avito.android.test.InteractionContext
+import com.avito.android.test.SimpleInteractionContext
+import com.avito.android.test.action.Actions
+import com.avito.android.test.action.ActionsDriver
+import com.avito.android.test.action.ActionsImpl
+import com.avito.android.test.espresso.EspressoActions
+import org.hamcrest.Matcher
+import org.hamcrest.Matchers
+
+class TextElement(interactionContext: InteractionContext) : ViewElement(interactionContext) {
+
+    constructor(matcher: Matcher<View>) : this(SimpleInteractionContext(matcher))
+
+    override val actions: TextElementActions = TextElementActionsImpl(interactionContext)
+
+    fun clickOnText(textToClick: String) = actions.clickOnText(textToClick)
+    fun clickOnLink() = actions.clickOnLink()
+    fun clickOnLink(position: Int) = actions.clickOnLink(position)
+}
+
+interface TextElementActions : Actions {
+    fun clickOnText(textToClick: String)
+    fun clickOnLink()
+    fun clickOnLink(position: Int)
```

Comment:
I have doubts about this API. 
In espresso we have simpler contract for view matchers - only one element must be found. In case of multiple similar element we must choose more precise matcher (clickOnText in this case).
I'd prefer to keep only two actions for now: clickOnText(String) and clickOnLink()